src/Apps/Order/Routes/Shipping/FulfillmentDetailsForm.tsx

```diff
@@ -0,0 +1,830 @@
+import {
+  Form,
+  Formik,
+  FormikErrors,
+  FormikHelpers,
+  FormikTouched,
+} from "formik"
+import { compact, isNil, omitBy, pick } from "lodash"
+import { FC, useCallback, useEffect, useState } from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import * as Yup from "yup"
+import { extractNodes } from "Utils/extractNodes"
+import { useFeatureFlag } from "System/useFeatureFlag"
+import { SavedAddressesFragmentContainer as SavedAddresses } from "Apps/Order/Components/SavedAddresses"
+
+const VALIDATION_SCHEMA = Yup.object().shape({
+  fulfillmentType: Yup.string().oneOf(Object.values(FulfillmentType)),
+
+  attributes: Yup.object()
+    .shape({
+      phoneNumber: yupPhoneValidator,
+      phoneNumberCountryCode: Yup.string().required("Phone number is required"),
+      name: Yup.string().required("Name is required"),
+    })
+    .when("fulfillmentType", {
+      is: FulfillmentType.SHIP,
+      then: schema =>
+        schema.shape({
+          addressLine1: Yup.string().required("Address is required"),
+          city: Yup.string().required("City is required"),
+          postalCode: postalCodeValidator,
+          region: Yup.string().required("Region is required"),
+          country: Yup.string().required("Country is required"),
+        }),
+    }),
+})
+
+interface PickupValues {
+  fulfillmentType: FulfillmentType.PICKUP
+  attributes: {
+    name: string
+    phoneNumber: string
+  }
+}
+
+export interface ShipValues {
+  fulfillmentType: FulfillmentType.SHIP
+  attributes: ShippingAddressFormValues & {
+    saveAddress: boolean
+    // TODO: This could go on the real address values, with a
+    // save_ here to signal when it should be updated.
+    // currently presence of this value means it will be updated
+    // only in exchange
+    addressVerifiedBy: AddressVerifiedBy | null
+  }
+}
+
+export type FulfillmentValues = ShipValues | PickupValues
+
+export interface FulfillmentDetailsFormProps {
+  active: boolean
+  onSubmit: (
+    values: FulfillmentValues,
+    formikHelpers: FormikHelpers<FulfillmentValues>
+  ) => void | Promise<any>
+  submitHandlerRef: React.MutableRefObject<(...args: any[]) => void>
+  me: FulfillmentDetailsForm_me$data
+  order: FulfillmentDetailsForm_order$data
+}
+
+const getShippingRestrictions = (
+  order: FulfillmentDetailsFormProps["order"]
+): {
+  lockShippingCountryTo: "EU" | string | null
+} => {
+  const firstArtwork = extractNodes(order.lineItems)[0]!.artwork!
+  const shippingCountry = firstArtwork.shippingCountry!
+  const domesticOnly = !!firstArtwork.onlyShipsDomestically
+  const euOrigin = !!firstArtwork.euShippingOrigin
+
+  const lockShippingCountryTo = domesticOnly
+    ? euOrigin
+      ? "EU"
+      : shippingCountry
+    : null
+  return { lockShippingCountryTo }
+}
+
+// Get initial values for the form using what may already be saved
+// on the order and the user's saved addresses.
+const getInitialValues = (
```

Comment:
The bulk of the shipping route (index.tsx) logic has been moved into this file, with the most important logic in this rather large helper. It's main role is to provide the initial values to formik when the address form loads.